src/xdotoolify.js

```diff
@@ -326,6 +326,45 @@ _Xdotoolify.prototype.autoType = function(selector, text, relpos, timeout) {
   }
   return this;
 };
+
+_Xdotoolify.prototype.sleepUntil = function(predicate, ms, immediate) {
```

Comment:
i think this interface could be better. interval between predicate calls and whether to make the first one immediately seem less relevant than a timeout before giving up. i.e. this should just be `function(predicate, timeout)` and the implementation something like:

```
while (true) {
  if (predicate()) {
    break;
  }
  if (exceededTimeout) {
    throw;
  }
  sleep(100);
}
```

which also is simpler and doesn't have a possibility for infinite loop like the current implementation.

Reply:
Fixed